### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by Yoka40**

### CODE REVIEW

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        vec.push_back(std::make_shared<int>(i));
+    }
+    return vec;
+}
+void print(const std::vector<std::shared_ptr<int>>& vec) {
+    for (const auto& el : vec) {
+        std::cout << *el << " ";
+    }
+}
+void add10(std::vector<std::shared_ptr<int>>& vec) {
```
Comment (piezol) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:24]: That's const reference as well

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    for (const auto& el : vec) {
+        if (el) {
+            *el += 10;
+        }
+    }
+}
+void sub10(std::vector<std::shared_ptr<int>>& vec) {
```
Comment (piezol) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:31]: That's const reference as well

File: module3/homework/passwordCheck/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,59 @@
...
+    switch (errCode)
+    {
+    case ErrorCode::Ok:
+        return "OK";
+    case ErrorCode::PasswordsDoesNotMatch:
+        return "Passwords does not match - check them";
+    case ErrorCode::PasswordNeedsAtLeastNineCharacters:
+        return "Password Needs at least 9 characters";
+    case ErrorCode::PasswordNeedsAtLeastOneNumber:
+        return "Min one sing must be numbr";
```
Comment (kyrtaq) [inline module3/homework/passwordCheck/validation.cpp:17]: why sing? :notes: it`s better to use whole words in error messages i think

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    for (size_t i = 0; i < count; ++i)
+    {
+        vec.push_back(std::make_shared<int>(i));
+    }
+    return vec;
+}
+void print(const std::vector<std::shared_ptr<int>>& vec) {
+    for (const auto& el : vec) {
+        std::cout << *el << " ";
+    }
```
Comment (kyrtaq) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:22]: be consistent in your mustache ( { } ) formatting :)

File: module3/homework/passwordCheck/test.cpp [hunk 5]
```diff
@@ -39,5 +39,9 @@ TEST(checkPasswordTests, returnsPasswordsDoesNotMatchForDifferentPasswords) {
 
 TEST(getErrorMessageTests, returnsErrorCodeAsString) {
     EXPECT_EQ(getErrorMessage(ErrorCode::Ok), "OK");
-    // Add other tests for getErrorMessage if you wish
+    EXPECT_EQ(getErrorMessage(ErrorCode::PasswordsDoesNotMatch), "Passwords does not match - check them");
+    EXPECT_EQ(getErrorMessage(ErrorCode::PasswordNeedsAtLeastNineCharacters), "Password Needs at least 9 characters");
+    EXPECT_EQ(getErrorMessage(ErrorCode::PasswordNeedsAtLeastOneNumber), "Min one sing must be numbr");
+    EXPECT_EQ(getErrorMessage(ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter), "Min one sing must be a special character.");
+    EXPECT_EQ(getErrorMessage(ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter), "Min one sing must be uppercase letter");
 }
```
Comment (m-kop) [inline module3/homework/passwordCheck/test.cpp:47]: newline missing

File: module3/homework/passwordCheck/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,59 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+#include <algorithm>
+
+
```
Comment (m-kop) [inline module3/homework/passwordCheck/validation.cpp:5]: remove additional empty line

File: module3/homework/vectorOfSharedPointers/vectorFunctions.hpp [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+std::vector<std::shared_ptr<int>> generate(int count);
+void print(const std::vector<std::shared_ptr<int>>& vec);
+void add10(std::vector<std::shared_ptr<int>>& vec);
+void sub10(int* const ptr);
+void sub10(std::vector<std::shared_ptr<int>>& vec);
```
Comment (Vicx95) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.hpp:10]:
```suggestion
void add10(const std::vector<std::shared_ptr<int>>& vec);
void sub10(int* const ptr);
void sub10(const std::vector<std::shared_ptr<int>>& vec);
```

File: module3/homework/passwordCheck/validation.hpp [hunk 8]
```diff
@@ -1,2 +1,17 @@
...
+const int minPassLength = 9;
+enum class ErrorCode {
+    Ok, 
+    PasswordNeedsAtLeastNineCharacters, 
+    PasswordNeedsAtLeastOneNumber,
+    PasswordNeedsAtLeastOneSpecialCharacter,
+    PasswordNeedsAtLeastOneUppercaseLetter,
+    PasswordsDoesNotMatch
+};
+std::string getErrorMessage(const ErrorCode& errCode);
```
Comment (m-kop) [inline module3/homework/passwordCheck/validation.hpp:14]: no need for &

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+{
+    std::vector<std::shared_ptr<int>> vec{};
+    if (count <= 0)
+    {
+        return vec;
+    }
+    vec.reserve(count);
+    for (size_t i = 0; i < count; ++i)
+    {
+        vec.push_back(std::make_shared<int>(i));
```
Comment (DamiXioR) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:15]: It's better to use emplace_back with objects

File: module3/homework/passwordCheck/validation.cpp (hunk 6 above)
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.cpp:5]: One enter is enough

File: module3/homework/passwordCheck/validation.cpp [hunk 10]
```diff
@@ -1,2 +1,59 @@
...
+        return "Passwords does not match - check them";
+    case ErrorCode::PasswordNeedsAtLeastNineCharacters:
+        return "Password Needs at least 9 characters";
+    case ErrorCode::PasswordNeedsAtLeastOneNumber:
+        return "Min one sing must be numbr";
+    case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter:
+        return "Min one sing must be a special character.";
+    case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter:
+        return "Min one sing must be uppercase letter";
+
```
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.cpp:22]: enter redundant

File: module3/homework/passwordCheck/validation.cpp [hunk 11]
```diff
@@ -1,2 +1,59 @@
...
+    default:
+        return "Something is bad";
+    }
+}
+bool doesPasswordsMatch(const std::string &firstPass, const std::string &secondPass)
+{
+    return firstPass == secondPass;
+}
+ErrorCode checkPassword(const std::string &firstPass, const std::string &secondPass)
```
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.cpp:31]: Enter between function

File: module3/homework/passwordCheck/validation.hpp [hunk 12]
```diff
@@ -1,2 +1,17 @@
-// TODO: I'm empty :) Put enum and function headers here.
-// Don't forget the header guard
\ No newline at end of file
+#pragma once
+
+#include<string>
+
+const int minPassLength = 9;
```
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.hpp:5]: constexpr will be better

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:24]: You don't modify vector, so use const.

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [1995 , 1996 , 1998 , 1119 , 1119],
"points": [7, 4, 5, 2, 2]
}

**Pull request closed without merging**